sdk/core/azure-core/src/main/java/com/azure/core/credential/AccessToken.java

```diff
@@ -19,7 +19,7 @@
      */
     public AccessToken(String token, OffsetDateTime expiresAt) {
         this.token = token;
-        this.expiresAt = expiresAt.minusMinutes(2); // 2 minutes before token expires
+        this.expiresAt = expiresAt;
```

Comment:
This will be a breaking change for developers not using the Identity library and using this class directly from the core for their custom credentials.

Reply:
I synced with @srnagar and he confirmed this is allowed in a minor version bump. It's not an API change but a behavioral change/fix. Users using this in conjunction with BearerTokenAuthenticationPolicy will experience the same behavior as the token cache now handles the 2 minutes default.